### PR TITLE
Decouple worker and master client

### PR DESCRIPTION
**Pull request by Yancey0623**
This PR implement a `MasterClient` class to decouple worker.py
c.f. step1 of #2149

### CODE REVIEW

File: elasticdl/python/worker/master_client.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from elasticdl.python.common.tensor_utils import serialize_ndarray
+
+
+class MasterClient:
+    """MasterClient provides some APIs connect with the master
+    service via gRPC call.
+
+    Usage:
+        channel = elasticdl.python.common.grpc_utils.build_channel(
+            "localhost:50001")
```
Comment (workingloong) [inline elasticdl/python/worker/master_client.py]:
```python
    channel = elasticdl.python.common.grpc_utils.build_channel(
            "localhost:50001"
    )
```

File: elasticdl/python/worker/master_client.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        mc = MasterClient(channel, work_id=0)
+        # get task unit from master service
+        mc.get_task(...)
+    """
+
+    def __init__(self, channel, worker_id):
+        """initialize a master client.
```
Comment (workingloong) [inline elasticdl/python/worker/master_client.py]: "Initialize a master client". Capitalize the first letter in the docstring.  There are the same questions in the following docstrings.

File: elasticdl/python/worker/master_client.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    def get_task(self, task_type=None):
+        """get task from master.
+
+        Args:
+            task_type: elasticdl_pb.TaskType
+            the training phase, c.f. /elasticdl/proto/elasticdl.proto
+
+        Returns:
+            the task unit assigned by master,
```
Comment (workingloong) [inline elasticdl/python/worker/master_client.py:54]: "the task assigned by master"

File: elasticdl/python/worker/master_client.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            )
+            res = elasticdl_pb2.Task()
+        return res
+
+    def report_task_result(self, task_id, err_msg, exec_counters=None):
+        """report task result to master.
+
+        Args:
+          task_id: int
+          the taskID assigned by master
```
Comment (workingloong) [inline elasticdl/python/worker/master_client.py]: the task ID assigned by master

**Review by QiJune — APPROVED** [review]
Excellent!

**Pull request merged**